backend/migrator/migration/prod/LATEST.sql

```diff
@@ -799,6 +799,31 @@ UPDATE
     ON activity FOR EACH ROW
 EXECUTE FUNCTION trigger_update_updated_ts();
 
+CREATE TABLE query_history (
+    id BIGSERIAL PRIMARY KEY,
+    row_status row_status NOT NULL DEFAULT 'NORMAL',
+    creator_id INTEGER NOT NULL REFERENCES principal (id),
+    created_ts BIGINT NOT NULL DEFAULT extract(epoch from now()),
+    updater_id INTEGER NOT NULL REFERENCES principal (id),
+    updated_ts BIGINT NOT NULL DEFAULT extract(epoch from now()),
+    database_id INTEGER NOT NULL REFERENCES db (id),
```

Comment:
Put database, statement in payload as well.

Database can be instances/x/databases/x format.

Reply:
It would be hard to filter by instance or database in the future🧐

Reply:
Or put in the payload, but keep using database_id?

Reply:
Oh, we also need the project id. I'm in favor of using database resource name instead of databaseid to avoid complexity of joins and foreign keys.

Reply:
I guess either way is fine, not a big deal. We still need to add project id column.

Reply:
how about

```sql
project_id INTEGER NOT NULL REFERENCES project (id),
database TEXT NOT NULL, -- the database resource name, for example, instance/{instance}/database/{database}
```

then we can still query by `database = xxx` or `database LIKE xxx` if needed in the future

Reply:
SGTM. The statement field can be kept either way. I am just in favor of schema-less structure. Wishing to support json index some day.